### PR TITLE
Delivery API: Add section about deeply nested JSON

### PR DESCRIPTION
**Pull request by kjac**
## Description

I have added a section about how to handle deeply nested JSON, if this is produced by the Delivery API as a consequence of deeply nested content structures.

My editor seems to have performed some clean-up of trailing white-spaces throughout the article. I guess this is based on the `.editorconfig` of this repo... might as well include those changes now rather than later 😄 

Cross-linking the origin for future reference: https://github.com/umbraco/Umbraco-CMS/issues/14610

## Type of suggestion

* [x] New content

## Product & version (if relevant)

v12

## Deadline (if relevant)

Not relevant.

### CODE REVIEW

File: 12/umbraco-cms/reference/content-delivery-api/README.md
Comment (sofietoft) [inline 12/umbraco-cms/reference/content-delivery-api/README.md:685]:
```suggestion
.NET imposes a limit on the depth of object nesting within rendered JSON. This is done in an effort to detect cyclic references. Learn more about it in [the official .NET API docs](https://learn.microsoft.com/en-us/dotnet/api/system.text.json.jsonserializeroptions.maxdepth)).

If the limit is exceeded, .NET will throw a `JsonException`.
```

File: 12/umbraco-cms/reference/content-delivery-api/README.md
Comment (sofietoft) [inline 12/umbraco-cms/reference/content-delivery-api/README.md:691]:
```suggestion
First, we have to add these `using` statements to `Startup.cs`:
```

**Review by sofietoft — COMMENTED** [review]
Thanks for the PR @kjac ! 💪 

I've added a couple of comments.

Also, regarding the formatting happening between lines 431 and 475 - I think we need to run those by @elit0451 . I seem to remember she had some issues with this when she initially wrote the article. It's a GitBook vs. GitHub thing.

File: 12/umbraco-cms/reference/content-delivery-api/README.md
Comment (alina-tincas) [inline 12/umbraco-cms/reference/content-delivery-api/README.md:685]: seems there is an extra ")" at the end 😄 
```suggestion
.NET imposes a limit on the depth of object nesting within rendered JSON. This is done in an effort to detect cyclic references. Learn more about it in [the official .NET API docs](https://learn.microsoft.com/en-us/dotnet/api/system.text.json.jsonserializeroptions.maxdepth).
```

**Review by alina-tincas — CHANGES_REQUESTED** [review]
Hi @kjac 👋 

I have a few suggestions to the changes you have made and was wondering if you can take a look at them and let me know your thoughts on them? 😊

**Comment by kjac** [comment]
Good stuff @alina-tincas ❤️ 

I've got to run some formatting past Elitsa on this one before it can be merged, so please don't merge it in just yet... I'll get back to you here.

**Comment by alina-tincas** [comment]
No worries, and sure thing 🙌

**Comment by jonat123** [comment]
@kjac great! 

As Alina is on vacation, i will merge in the PR :D

**Review by jonat123 — APPROVED** [review]

**Pull request merged**
